### DISCUSSION TITLE
In the  built in search, it doesn't go to the first finding closest to where the cursor is at.

### DISCUSSION THREAD

**Discussion by aminasharaf** [stephanrauh/ngx-extended-pdf-viewer#2514] [Q&A]
In the  built in search, it doesn't go to the first finding closest to where the cursor is at. Is there a way to achieve this?

**Comment by stephanrauh** [comment]
Which version of ngx-extended-pdf-viewer are your using? I'm asking because I've been working a lot on the find controller during the last two or three weeks.

Generally speaking: I'm surprised you say it doesn't go the the closest finding. It doesn't go back, but at least it should go to the closest finding after the cursor. I thought it works just like that. Can you check if https://mozilla.github.io/pdf.js/web/viewer.html works differently than ngx-extended-pdf-viewer, please?

**Comment by aminasharaf** [comment]
I am using the version 20.0.2. https://mozilla.github.io/pdf.js/web/viewer.html also works similar to ngx-extended-pdf-viewer.

**Reply by stephanrauh** [reply]
So it's probably an issue with the base library. Please report it at https://github.com/mozilla/pdf.js/issues. When you do, please create a video of your bug. Ideally, the bug should show either in the native PDF viewer of Firefox or on https://mozilla.github.io/pdf.js/web/viewer.html.

**Comment by aminasharaf** [comment]
Sorry for the misunderstanding. Actually, on this page, https://mozilla.github.io/pdf.js/web/viewer.html, it works properly as it does go to the first finding closest to where the cursor is located.

**Reply by stephanrauh** [reply]
OK. Then creating a video is even more important. When I try the search, everything works fine. Or is there a difference between my showcase (https://pdfviewer.net/extended-pdf-viewer/simple) and your application?

**Reply by stephanrauh** [reply]
Please update to version 21.3.4. It solves a bug that causes the viewer to jump to the top of the page instead of scrolling the selected search result into view. Maybe that's what you've been reporting.

At the time of writing, I haven't published the version, but it's probably going to go live this week-end.

**Reply by aminasharaf** [reply]
Thankyou. I will check it out.
